### PR TITLE
feat: React Native 0.73 compatibility

### PR DESCRIPTION
**Pull request by crherman7**
## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist before requesting a review

- [ ] A self-review of my code has been completed
- [ ] Tests have been added / updated if required
- [ ] Documentation has been updated to reflect these changes

### CODE REVIEW

File: packages/cli-kit/src/lib/path/index.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  '0.72': pathRN72,
+  '0.73': pathRN73,
+};
+
+// Get the React Native version
+const reactNativeVersion = getReactNativeVersion();
+
+// Ensure the version exists in the profiles object
+if (!(reactNativeVersion in profiles)) {
+  throw new Error(`Unsupported React Native version: ${reactNativeVersion}`);
```
[the remark this answers is not in the document]
Comment (crherman7) [reply packages/cli-kit/src/lib/path/index.ts:23]: `getReactNativeVersion` error message serves different purpose than the path error messaging - the former is all about making sure the package is installed and version is parse-able while the latter determines if the version profile exists for the path module. Not sure where else you would throw the error as you would want to know what's wrong immediately - and the function basically needs to run at top level to not interrupt how the path module is used currently which is just static based except for a few paths. Idk thoughts?

File: packages/cli/src/actions/dependencies.ts [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+
+    const executedProfiles: string[] = [];
+
+    /**
+     * Verifies and updates a dependency based on the given profile.
+     * @param {string} dependency - The name of the dependency.
+     * @param {Profile} profile - The profile specifying the version and other details for the dependency.
+     */
+    function verifyDependency(dependency: string, profile: Profile) {
+      if (executedProfiles.includes(dependency)) return;
```
[the remark this answers is not in the document]
Comment (crherman7) [reply packages/cli/src/actions/dependencies.ts:38]: can you give an example where dependency would have multiple profiles?

File: packages/cli/src/actions/dependencies.ts [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+
+      executedProfiles.push(dependency);
+
+      const version =
+        pkg.content?.[profile.devOnly ? 'devDependencies' : 'dependencies']?.[
```
[the remark this answers is not in the document]
Comment (crherman7) [reply packages/cli/src/actions/dependencies.ts:43]: that would probably be more ideal

File: packages/cli/src/actions/dependencies.ts [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+    pkg.update({
+      dependencies: {
+        ...(pkg.content.dependencies as any),
+      },
+      devDependencies: {
+        ...(pkg.content.devDependencies as any),
+      },
+    });
+
+    await pkg.save();
```
[the remark this answers is not in the document]
Comment (crherman7) [reply packages/cli/src/actions/dependencies.ts:120]: yeah fair point

File: packages/cli-kit/src/lib/path/index.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (NickBurkhartBB) [reply packages/cli-kit/src/lib/path/index.ts:23]: initial thoughts are just wrapping it and putting it as like a prerequisite for the specific commands that require the check on react-native version

File: packages/cli/src/actions/dependencies.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (NickBurkhartBB) [reply packages/cli/src/actions/dependencies.ts]: didn't have an example, just noticed the executed list only used dependency for key and wondered if we hit with this function with a different value. It was mostly a note to double check entry points.

**Review by NickBurkhartBB — APPROVED** [review]

**Pull request merged**